src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -32,19 +32,25 @@ public static class TagHelperDescriptorFactory
         /// Creates a <see cref="TagHelperDescriptor"/> from the given <paramref name="type"/>.
         /// </summary>
         /// <param name="assemblyName">The assembly name that contains <paramref name="type"/>.</param>
+        /// <param name="prefix">The prefix to the created <see cref="TagHelperDescriptor"/>s.</param>
         /// <param name="type">The type to create a <see cref="TagHelperDescriptor"/> from.</param>
         /// <returns>A <see cref="TagHelperDescriptor"/> that describes the given <paramref name="type"/>.</returns>
-        public static IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName, Type type)
+        public static IEnumerable<TagHelperDescriptor> CreateDescriptors(
+            string assemblyName, 
+            string prefix,
+            Type type)
         {
             var tagNames = GetTagNames(type);
             var typeName = type.FullName;
             var attributeDescriptors = GetAttributeDescriptors(type);
 
             return tagNames.Select(tagName =>
-                new TagHelperDescriptor(tagName,
-                                        typeName,
-                                        assemblyName,
-                                        attributeDescriptors));
+                new TagHelperDescriptor(
+                    prefix,
+                    tagName,
+                    typeName,
+                    assemblyName,
+                    attributeDescriptors));
```

Comment:
so much simpler and less-invasive if you create an unchanged `TagHelperDescriptor` with `prefix + tagName` for the `tagName` argument